sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpMethodDeclarationTracker.cs

```diff
@@ -51,7 +51,8 @@ public CSharpMethodDeclarationTracker(IAnalyzerConfiguration analysisConfigurati
                 }
 
                 var methodDeclaration = context.MethodSymbol.DeclaringSyntaxReferences
-                    .Select(r => (MethodDeclarationSyntax)r.GetSyntax())
+                    .Select(r => r.GetSyntax())
+                    .OfType<BaseMethodDeclarationSyntax>()
```

Comment:
The property getters and setters are also methods, hence we filter only "normal" methods, constructors and destructors here.